website/versioned_docs/version-v1.x/contracts.md

```diff
@@ -15,142 +15,171 @@ The main contracts are presented and explained below.
 
-After this, the contracts will be stored to state, the current time taken and then the contract will perform a simple sanity check to ensure that the Poseidon hash libraries were linked successfully.
+After this, the contracts will be stored to state, and then the contract will perform a simple sanity check to ensure that the Poseidon hash libraries were linked successfully.
 
-Next, we have the `signUp` function, which allows users to `signUp` using a `SignUpGatekeeper` contract. This contract can use any mean necessary to gatekeep access to MACI's polls. For instance, only wallets with access to a specific ERC721 token can be allowed to sign up. Please note that this function can only be called after the contract is initialized (thanks to the `afterInit` modifier).
+Next, we have the `signUp` function, which allows users to `signUp`, as long as they pass the conditions set in the `SignUpGatekeeper` contract. This contract can use any mean necessary to gatekeep access to MACI's polls. For instance, only wallets with a specific ERC721 token can be allowed to sign up.
 
 This function does the following:
 
 - checks that the maximum number of signups has not been reached. As of now, this will be $5 ** 10 - 1$ due to circuit limitations.
-- checks that the provided public key is within the allowed boundaries
-- increases the number of signups
-- registers the user using the sign up gatekeeper contract. It is important that whichever gatekeeper is used, it reverts if a user tries to sign up twice.
-- calls the voice credit proxy to retrieve the number of allocated voice credits for the calling account
-- hashes the voice credits alongside the calling address and the current time
+- checks that the provided public key is a valid baby-jubjub point
+- increases signups counter
+- registers the user using the sign up gatekeeper contract. It is important that whichever gatekeeper is used, this reverts if a user tries to sign up twice or the conditions are not met (i.e returning false is not enoguh)
+- calls the voice credit proxy to retrieve the number of allocated voice credits allocated to this voter
+- hashes the voice credits alongside the user's MACI public key and the current time
 - enqueues this hashed data into the `stateAq` contract
 
 ```javascript
 function signUp(
     PubKey memory _pubKey,
     bytes memory _signUpGatekeeperData,
     bytes memory _initialVoiceCreditProxyData
-) public afterInit {
-    // The circuits only support up to (5 ** 10 - 1) signups
-    require(
-        numSignUps < STATE_TREE_ARITY**stateTreeDepth,
-        "MACI: maximum number of signups reached"
-    );
-
-    require(
-        _pubKey.x < SNARK_SCALAR_FIELD && _pubKey.y < SNARK_SCALAR_FIELD,
-        "MACI: _pubKey values should be less than the snark scalar field"
-    );
+) public virtual {
+    // prevent new signups until we merge the roots (possible DoS)
+    if (subtreesMerged) revert SignupTemporaryBlocked();
```

Comment:
we also check if the subtree is merged or not, maybe should add this to description?